src/util.js

```diff
@@ -62,14 +62,36 @@ module.exports = {
 
     return true;
   },
-  game({seats, type, cube}) {
-    assert(typeof seats === "number", "seats should be a number");
-    assert(2 <= seats && seats <= 100, "seats' number should be between 2 and 100");
+  game({seats, type, sets, cube, isPrivate, fourPack, modernOnly = true, totalChaos = true}) {
     assert(["draft", "sealed", "cube draft", "cube sealed", "chaos"].includes(type),
       "type can be draft, selaed, cube draft or cube sealed");
+    assert(typeof isPrivate === "boolean", "isPrivate must be a boolean");
+    assert(typeof seats === "number", "seats must be a number");
+    assert(2 <= seats && seats <= 100, "seats' number must be between 2 and 100");
 
-    if (/cube/.test(type))
+    switch(type) {
+    case "draft":
+    case "sealed":
+      assert(Array.isArray(sets), "sets must be an array");
+      sets.forEach(set =>
+        assert(Sets[set] !== undefined, `set ${set} is invalid or does not exist`));
+      if("sealed" === type) {
+        assert(typeof fourPack === "boolean", "fourPack must be a boolean");
+        assert(!fourPack || sets.length < 4, "sets must be at least 4");
+      } else {
+        assert(sets.length == 3, "sets length must be 3");
+      }
+      break;
+    case "cube draft":
+    case "cube sealed":
+      assert(typeof cube === "object", "cube must be an object");
```

Comment:
Cubes are currently a newline delimited string

Reply:
Hi @dev-id ,

Correct me if I'm wrong, but currently the cube is an object made of "list", "cards", "packs". 
If you look at the cube() function, create just before creating the game from frontend: 
https://github.com/dr4fters/dr4ft/blob/775e20cf31b7dada27bc2c702fb54559d4f6957b/public/src/cards.js#L154-L155

https://github.com/dr4fters/dr4ft/blob/775e20cf31b7dada27bc2c702fb54559d4f6957b/public/src/cards.js#L353-L369

It looks like we send an object with "list", "cards" and "packs".

Reply:
Ahh you're correct. Was thinking of `list`